### ISSUE TITLE
Thetis non-reponsive

### ISSUE THREAD

**Issue by mdblack98** [ramdor/Thetis#242]
**Describe the bug**
As of ce8f43294354f3cb0d26b78adb31edbf2d1213ba was running overnight.
Tried to just click on the window to bring it to front -- no joy.  New CAT control connection did not  work.
Spectrum still updating.  Could not  close it....had to kill it.
I'll build a debug version and see if it reproduces so I can attach to it.

**To Reproduce**
Steps to reproduce the behaviour:
1. Run Thetis overnight.
2. Non-reponsive
3
**Expected behaviour**
Should continue being responsive :-)

**Screenshots**
N/A

**Desktop (please complete the following information):**
Windows  11 25982.1000

**Additional context**
N/A

**Comment by ramdor** [comment]
did anything on pc sleep
did screen sleep
did windows update anything
did monitor get turned off

**Comment by ramdor** [comment]
please include errorlog.txt, better yet, use button in setup general->h/w select, to create a zip of it all. Cheers.

![image](https://github.com/ramdor/Thetis/assets/3124967/475d7d87-b232-45c6-9425-6713ab8f930c)

**Comment by mdblack98** [comment]
Nope --none of that.  I've been running the new builds most every day but the big updates in the last couple of days seems to have caused this.
I did a debug build and the GUI is  non-responsive there  in pretty short order (like 30 minutes or so).
Seems hard to catch where the loop is occurring.
I'll do a git bisect  to try and narrow it down although I can't seem to cause it to happen -- it just happens on it's own.
Presumably the event pump is locked up.
 

    On Wednesday, November 1, 2023 at 10:54:09 AM CDT, Richie ***@***.***> wrote:  
 
 


did anything on pc sleep
did screen sleep
did windows update anything
did monitor get turned off

**Comment by mdblack98** [comment]
https://www.dropbox.com/scl/fi/hhfch82jlfwf2zvpoainr/11-01-23_111357_database_logs.zip?rlkey=peq6etrob762ss5qbksbdbd0e&dl=0


 Mike W9MDB
    On Wednesday, November 1, 2023 at 11:06:23 AM CDT, Richie ***@***.***> wrote:  
 
 


please include errorlog.txt, better yet, use button in setup general->h/w select, to create a zip of it all. Cheers.

**Comment by ramdor** [comment]
was it doing anything during this time? cat comms via tcpip, tci?  I just ran memory profiler for about an hour and no loss to speak of.

**Comment by mdblack98** [comment]
Just idling.  I had FLRig hooked  to it and Log4OM was using rigctld to talk to FLRig  so may a little busier on the CAT data.No transmitting.
This morning's lockup was just FLRig by itself.
Right now I have a few hours of run-time on the debug build and it's OK...but we'll see.  Debug build did lock up this morning.
I would've been clicking on other windows on the screen but not on Thetis -- first click on it was locked up.

Mike W9MDB



 

    On Wednesday, November 1, 2023 at 12:52:55 PM CDT, Richie ***@***.***> wrote:  
 
 


was it doing anything during this time? cat comms via tcpip, tci? I just ran memory profiler for about an hour and no loss to speak of.

**Comment by ramdor** [comment]
and what connection method was being used to connect to thetis,  serial, tcpip, tci?

**Comment by mdblack98** [comment]
Serial

 

    On Wednesday, November 1, 2023 at 04:43:50 PM CDT, Richie ***@***.***> wrote:  
 
 


and what connection method was being used to connect to thetis, serial, tcpip, tci?

**Comment by ramdor** [comment]
any tcpip cat connections as well? or just the single serial?

**Comment by mdblack98** [comment]
Just one serial connection.  Others  enabled but not used.

 

    On Wednesday, November 1, 2023 at 04:46:25 PM CDT, Richie ***@***.***> wrote:  
 
 


any tcpip cat connections as well? or just the single serial?

**Comment by mdblack98** [comment]
Previous to the big updates in the last few days never saw this problem.Wish I could repeat it more easily....but have no idea what causes it.
Maybe  I can look at the thread and find one chewing up CPU or maybe in some wait state.

 

    On Wednesday, November 1, 2023 at 04:46:25 PM CDT, Richie ***@***.***> wrote:  
 
 


any tcpip cat connections as well? or just the single serial?

**Comment by ramdor** [comment]
try latest master. Strange things were being done in serial listener

**Comment by W2PA** [comment]
This, or something similar, was happening for me too, after pulling changes yesterday (11/1) morning.  Thetis would start up seemingly fine, hearing signals, display working - but the MON, TUN, and MOX buttons were greyed out and the console did not respond to mouse events; I had to kill it to terminate.  I reverted to a version from last weekend and all was fine again.  

I just pulled recent changes this morning and the weird behavior is now gone.  I'll continue to let it run and test it.
